addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1594,7 +1594,7 @@ const Wysiwyg = Widget.extend({
         }
         const fonts = this.odooEditor.execCommand('applyColor', color, eventName === 'foreColor' ? 'color' : 'backgroundColor', this.lastMediaClicked);
 
-        if (!this.lastMediaClicked) {
+        if (!this.lastMediaClicked && fonts) {
```

Comment:
If the idea is to avoid any traceback:
execCommand('applyColor') can return an empty array, which is truthy, but will make `last` be undefined when trying to read index -1. So might as well check if fonts.length is > 0.

Reply:
As a side note, if `execCommand` returns "UNBREAKABLE" or "UNREMOVABLE", it would still crash. But I guess that's the kind of error we do not want to be silenced, as it points to a problem somewhere else in the code.

Reply:
@cammarosano @dmo-odoo my idea is to avoid traceback using font because the problem lies in applyRawCommand. From applyRawCommand it returns false because there is a condition if sel.anchorNode not in editable then it returns false and our selection stuck on hex Input box in colorpicker. And when we try to adjust selection for such cases it gives some error, example, When we try to color a text it wont get color and in some cases when we click on hex input box then it closes the colorpicker.

Reply:
Not completely sure I understood it all, but...
Yes, `applyRawCommand` returns `false` if the selection in not in the editable. And you should not change it. Instead, the caller (`_processAndApplyColor`) should handle it. I would go for a simple
`if (!fonts) return;`

Just as a reminder, I'm glad to participate in any discussion, but I'm not a reviewer. @dmo-odoo has the final word!